rest/model/patch.go

```diff
@@ -314,8 +314,13 @@ func (apiPatch *APIPatch) buildModuleChanges(p patch.Patch, identifier string) {
 		htmlLink := fmt.Sprintf("%s/filediff/%s?patch_number=%d", apiURL, *apiPatch.Id, patchNumber)
 		rawLink := fmt.Sprintf("%s/rawdiff/%s?patch_number=%d", apiURL, *apiPatch.Id, patchNumber)
 		fileDiffs := []FileDiff{}
-		for _, file := range modPatch.PatchSet.Summary {
-			diffLink := fmt.Sprintf("%s/filediff/%s?file_name=%s&patch_number=%d", apiURL, *apiPatch.Id, url.QueryEscape(file.Name), patchNumber)
+		commitNumber := 0
+		prevCommitMsg := ""
+		for i, file := range modPatch.PatchSet.Summary {
+			if i > 0 && file.Description != prevCommitMsg {
+				commitNumber++
+			}
+			diffLink := fmt.Sprintf("%s/filediff/%s?file_name=%s&patch_number=%d&commit_number=%d", apiURL, *apiPatch.Id, url.QueryEscape(file.Name), patchNumber, commitNumber)
```

Comment:
is there a reason to not just always append i as the commit number? Will these always be in order where we only need to consider the exact previous commit message (as opposed to the set of previous ones)?

Reply:
Sure, we might as well do that. This route is just reading from the full diff which will be the same order every time.